linked_list/LinkedList.c

```diff
@@ -0,0 +1,179 @@
+		nw_node->next = NULL;
+		*start = nw_node;
+	}
+	else {
+		nw_node->next = *start;	// Else add address of first node to nw_node and make it first node i.e.'start'
+		*start = nw_node;
+	}
+}
+
+//  Adding at last
+void addLast(struct Node **start,int data) {
+	struct Node* nw_node = (struct Node*) malloc(sizeof(struct Node));	// nw_node a node to be added
+	struct Node* temp;		//  'temp' temporary node to traverse to last
+	nw_node->next = NULL;		//  Last element must have null value in its next variable
+	nw_node->d = data;		//  data
+	if (*start == NULL) {
+		*start = nw_node;
+	}
+	else {
+		temp = *start;
+		while (temp->next != NULL) {	// Loop until we get last node
+			temp = temp->next;
+		}
+		temp->next = nw_node;		// Assiging address of nw_node to next variable of last element of list
+	}
+}
+
+//  Adding at given position
+void addAtPos(struct Node **start,int data, int pos) {
+	struct Node* nw_node = (struct Node*) malloc(sizeof(struct Node));	// nw_node a node to be added,
+	struct Node* temp;		// 'temp' temporary node to traverse to last.
+	nw_node->d = data;		// data
+	if (*start == NULL) {
+		*start = nw_node;
+	}
+	else {
+		int i = 1;
+		temp = *start;
+		while (temp->next != NULL && i < pos-1) {	// Loop we reach to the position or last element of list is encountered
+			temp = temp->next;
+			i++;
+		}
+		if (i != pos)
+			printf("Invalid Index\n\n");
+		else {
+		    nw_node->next = temp->next;		// Placing address of next node to nw_node->next
+		    temp->next = nw_node;		// Assiging address of nw_node in list after temp node.
+		}
+	}
+}
+
+//  Deleting first element
+void deleteFirst(struct Node **start) {
+	if (*start == NULL)
+		printf("List is Empty.\n\n");
+	else {
+		*start = (*start)->next;		// Shifting 'start' to second node
+	}
+}
+
+//  Deleting last element
+void deleteLast(struct Node **start) {
+	if (*start == NULL)
+		printf("List is Empty.\n");
+	else {
+		struct Node* temp = *start;		// temp node to go through each node
+		while (temp->next->next != NULL)	// Loop until second last element is encountered
+			temp = temp->next;
+		temp->next = NULL;			// Removing last element
+	}
+}
+
+//  Delete element at given position
+void deleteAtpos(struct Node **start,int pos) {
+	if (*start == NULL) {
+		printf("List is Empty.\n\n");
+	}
+	else {
+		struct Node* temp = *start;		// temp node to go through each node
+		int i = 1;
+		while (temp->next->next != NULL && i < pos-1) {	// until we reach to the position or last element of list is encountered
+			temp = temp->next;
+			i++;
+		}
+		if (i != pos)
+			printf("Invalid Index\n\n");
+		else
+		    temp->next = temp->next->next;	// Replacing address of position 'pos' with  node next to 'pos'
+	}
+}
+
+//  To traverse whole list
+void traverse(struct Node **start) {
+    printf("Linked List\n");
+	if (*start == NULL)
+		printf("List is Empty\n");
+	else {
+		struct Node* temp = *start;		// temp node to go through each node
+		while (temp->next != NULL) {
```

Comment:
Actually instead of try catch I used it. Otherwise in case of traversing of empty list it will give null pointer exception.